Model/DataSerialization.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GameOfLife
+{
+    /// <summary>
+    /// Logic behind saving data to a file.
+    /// </summary>
+    public class DataSerialization
+    {
+        /// <summary>
+        /// Method to save game data to file.
+        /// </summary>
+        /// <typeparam name="T">Object type.</typeparam>
+        /// <param name="serializableObject">Object to save to file.</param>
+        /// <param name="fileName">File path to save data to.</param>
+        public void SerializeObject<T>(T serializableObject, string fileName)
+        {
+            if (File.Exists(fileName))
```

Comment:
Check whether deletion of file is needed, because later you are not opening file but creating it.
It looks that pre existing file fill be overwritten.

Reply:
Removed the `if` statement deleting the file.